op-chain-ops/genesis/layer_two.go

```diff
@@ -46,3 +54,74 @@ func BuildL2DeveloperGenesis(config *DeployConfig, l1StartBlock *types.Block) (*
 
 	return db.Genesis(), nil
 }
+
+// BuildL2MainnetGenesis will build an L2 Genesis suitable for a Superchain mainnet that does not
+// require a pre-bedrock migration & supports optional governance token predeploy.
```

Comment:
It would be great to have a spec of sorts in this godoc so that we can be sure that the impl is correct.
- Proxies in the address space `[0x4200000000000000000000000000000000000000, 0x4200000000000000000000000000000000000800)` for predeploys
- All predeploy proxies owned by the `ProxyAdmin`
- Predeploys as per the [spec](https://github.com/ethereum-optimism/optimism/blob/develop/specs/predeploys.md)
  - except for no  `LegacyERC20ETH` at `0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000`
  - optional governance token at `0x4200000000000000000000000000000000000042`